tests/utils/test_cache.py

```diff
@@ -192,3 +195,70 @@ def test_cachy_compatibility(
 
     assert cachy_file_cache.get("key3") == test_str
     assert cachy_file_cache.get("key4") == test_obj
+
+
+def test_missing_cache_file(repository_cache_dir: Path, mocker: MockerFixture) -> None:
+    cache = FileCache(repository_cache_dir / "cache")
+
+    cache.put("key1", "value")
+
+    key1_path = (
+        repository_cache_dir
+        / "cache"
+        / "81/74/09/96/87/a2/66/21/8174099687a26621f4e2cdd7cc03b3dacedb3fb962255b1aafd033cabe831530"  # noqa: E501
+    )
+    assert key1_path.exists()
+    key1_path.unlink()  # corrupt cache by removing a key file
+
+    assert cache.get("key1") is None
+
+
+def test_missing_cache_path(repository_cache_dir: Path, mocker: MockerFixture) -> None:
+    cache = FileCache(repository_cache_dir / "cache")
+
+    cache.put("key1", "value")
+
+    key1_partial_path = repository_cache_dir / "cache" / "81/74/09/96/87/a2/"
+    assert key1_partial_path.exists()
+    shutil.rmtree(
+        key1_partial_path
+    )  # corrupt cache by removing a subdirectory containting a key file
+
+    assert cache.get("key1") is None
+
+
+@pytest.mark.parametrize(
+    "corrupt_payload",
+    [
+        "",  # empty file
+        b"\x00",  # null
+        "99999999",  # truncated file
+        '999999a999"value"',  # corrupt lifetime
+        b'9999999999"va\xd8\x00"',  # invalid unicode
+        "fil3systemFa!led",  # garbage file
+    ],
+)
+def test_detect_corrupted_cache_key_file(
+    corrupt_payload: str | bytes, repository_cache_dir: Path, mocker: MockerFixture
+) -> None:
+    cache = FileCache(repository_cache_dir / "cache")
+
+    cache.put("key1", "value")
+
+    key1_path = (
+        repository_cache_dir
+        / "cache"
+        / "81/74/09/96/87/a2/66/21/8174099687a26621f4e2cdd7cc03b3dacedb3fb962255b1aafd033cabe831530"  # noqa: E501
+    )
+    assert key1_path.exists()
+
+    # original content: 9999999999"value"
+
+    write_modes = {"str": "w", "bytes": "wb"}
+    with open(key1_path, write_modes[corrupt_payload.__class__.__name__]) as f:
```

Comment:
This should work if it's changed to `write_modes = {str: "w", bytes: "wb"}` and `type(corrupt_payload)` right? That seems a bit more readable.